src/logic/wallets/store/reducer/index.ts

```diff
@@ -0,0 +1,61 @@
+import { Action, handleActions } from 'redux-actions'
+
+import { ChainId } from 'src/config/chain.d'
+import { PROVIDER_ACTIONS } from 'src/logic/wallets/store/actions'
+
+export type ProvidersState = {
+  name: string
+  hardwareWallet: boolean
+  smartContractWallet: boolean
+  network: ChainId
+  account: string
+  available: boolean
+  ensDomain: string
+  loaded: boolean
+}
+
+export type ProviderWalletPayload = Pick<ProvidersState, 'name' | 'hardwareWallet' | 'smartContractWallet'>
+export type ProviderNetworkPayload = ProvidersState['network']
+export type ProviderAccountPayload = ProvidersState['account']
+export type ProviderEnsPayload = ProvidersState['ensDomain']
+
+export type ProviderPayloads =
+  | ProviderWalletPayload
+  | ProviderAccountPayload
+  | ProviderNetworkPayload
+  | ProviderEnsPayload
+
+const initialProviderState: ProvidersState = {
+  name: '',
+  hardwareWallet: false,
+  smartContractWallet: false,
+  account: '',
+  network: '',
+  ensDomain: '',
+  available: false,
+  loaded: false,
```

Comment:
Do we need both of these properties?

Reply:
There's an [issue to tackle removing these](https://github.com/gnosis/safe-react/issues/3338) as it is a bit out of scope.

Reply:
Your comment regarding the factory relates to the handling of these two flags to. It extracts the logic.

Reply:
(the `providerFactory` here is not the one I posted as an example)

Reply:
Confirmed as resolved by @mikheevm